src/Build/Construction/ProjectRootElement.cs

```diff
@@ -1852,53 +1852,60 @@ internal void MarkAsExplicitlyLoaded()
             IsExplicitlyLoaded = true;
         }
 
+#nullable enable
+
         /// <summary>
         /// Creates and returns a list of <see cref="ProjectImportElement"/> nodes which are implicitly
         /// referenced by the Project.
         /// </summary>
         /// <param name="currentProjectOrImport">Current project</param>
-        /// <returns>An <see cref="IEnumerable{SdkReference}"/> containing details of the SDKs referenced by the project.</returns>
+        /// <returns>A <see cref="List{ProjectImportElement}"/> of implicit &lt;Import /&gt; elements for the SDKs referenced by the project.</returns>
         internal List<ProjectImportElement> GetImplicitImportNodes(ProjectRootElement currentProjectOrImport)
         {
             var nodes = new List<ProjectImportElement>();
 
-            string sdkAttribute = Sdk;
+            var sdkAttribute = Sdk;
             if (!string.IsNullOrWhiteSpace(sdkAttribute))
             {
-                foreach (var referencedSdk in ParseSdks(sdkAttribute, SdkLocation))
+                var location = SdkLocation;
+                var origin = new SdkReferenceOrigin(location, location, location);
+
+                foreach (var sdk in sdkAttribute.Split(MSBuildConstants.SemicolonChar).Select(i => i.Trim()))
                 {
-                    nodes.Add(ProjectImportElement.CreateImplicit("Sdk.props", currentProjectOrImport, ImplicitImportLocation.Top, referencedSdk, this));
-                    nodes.Add(ProjectImportElement.CreateImplicit("Sdk.targets", currentProjectOrImport, ImplicitImportLocation.Bottom, referencedSdk, this));
+                    if (!SdkReference.TryParse(sdk, out var reference))
+                    {
+                        ProjectErrorUtilities.ThrowInvalidProject(location, "InvalidSdkFormat", sdkAttribute);
+                        break;
```

Comment:
Explicit (unreachable) break to make control flow a bit more correct for the code analysis tools. I don't like these asserts in MSBuild, but I kept them in existing code.